databricks/sdk/credentials_provider.py

```diff
@@ -49,12 +69,34 @@ def inner(func: Callable[['Config'], HeaderFactory]) -> CredentialsProvider:
 
         @functools.wraps(func)
         def wrapper(cfg: 'Config') -> Optional[HeaderFactory]:
+            for attr in require:
+                getattr(cfg, attr)
+                if not getattr(cfg, attr):
+                    return None
+            return func(cfg)
+
+        wrapper.auth_type = lambda: name
+        return wrapper
+
+    return inner
+
+
+def oauth_credentials_provider(name: str, require: List[str]):
+    """ Given the function that receives a Config and returns an OauthHeaderFactory,
+    create an OauthCredentialsProvider with a given name and required configuration
+    attribute names to be present for this function to be called. """
+
+    def inner(func: Callable[['Config'], OAuthHeaderFactory]) -> OauthCredentialsProvider:
+
+        @functools.wraps(func)
+        def wrapper(cfg: 'Config') -> Optional[OAuthHeaderFactory]:
             for attr in require:
                 if not getattr(cfg, attr):
                     return None
             return func(cfg)
 
         wrapper.auth_type = lambda: name
+        wrapper.oauth_token = lambda cfg: wrapper(cfg).oauth_token()
```

Comment:
This is kind of strange looking, but I think it does work. We've defined OauthCredentialsProvider as a metaclass, but maybe we could define an instance of it here. I'm worried because I'm not sure the Python type checking is powerful enough to verify that we're setting the correct attribute on the inner classes. Better might be to define OauthCredentialsProvider as a proper class with an `__init__` method, where these functions can be passed in via constructor. That way, we get loud failures if we forget to specify some element of the credential provider.